### ISSUE TITLE
Union and Subtract are not working properly on Cube & Sphere

### ISSUE THREAD

**Issue by sam016** [samalexander/three-csg-ts#32]
**Describe the bug**
I am trying the Union and Subtract operation on Cube and Sphere, and it's not working as expected. 

**To Reproduce**

```ts

const getCube = () => {
  const geometry = new THREE.BoxGeometry(2, 2, 2);
  const material = new THREE.MeshPhongMaterial({
    color: 0x156289,
    emissive: 0x072534
  });
  const cube = new THREE.Mesh(geometry, material);
  return cube;
};

const getSphere = () => {
  const geometry = new THREE.SphereGeometry();
  const material = new THREE.MeshPhongMaterial({
    color: 0x156289,
    emissive: 0x072534
  });
  const sphere = new THREE.Mesh(geometry, material);
  sphere.position.set(1, 1, 1);
  return sphere;
};

const op = (
  type: "union" | "subtract" | "intersect",
  meshA: THREE.Mesh,
  meshB: THREE.Mesh
) => {
  meshA.updateMatrix();
  meshB.updateMatrix();

  switch (type) {
    case "union":
      return CSG.union(meshA, meshB);
    case "subtract":
      return CSG.subtract(meshA, meshB);
    case "intersect":
      return CSG.intersect(meshA, meshB);
    default:
      throw new Error("Invalid operation type");
  }
};

scene.add(op("union", getCube(), getSphere()));
```
Here is the sandbox for the same: https://codesandbox.io/s/three-csg-sphere-issue-zxnfq

**Expected behavior**
I was expecting it to give the correct union & subtract without any transparency.

**Screenshots**
Here is how it looks like:
![Union & Subtract](https://i.imgur.com/Tu1s5f2.gif)

**Additional context**

**Comment by altavir** [comment]
I have a similar problem:
![image](https://user-images.githubusercontent.com/9514268/125166627-35005600-e1a5-11eb-80be-a0e3c5df6c8a.png)
The second solid is just missing.

**Comment by altavir** [comment]
Still there on 3.1.5 and three-js 0.130.1

**Comment by samalexander** [comment]
Fix released in v3.1.6
